### PR TITLE
Made improvements to download cache

### PR DESCRIPTION
**Pull request by jhemelhof**
New pull request, this time cleaned up. I've made some improvements to the download cache, more specific solved the first two "issues" raised in #7571.

If there's anything that needs to be changed/improved feel free to tell it.

### CODE REVIEW

**Comment by alebcay** [comment]
Any update on this?

File: lib/hbc/cli/cleanup.rb [hunk 1]
```diff
@@ -6,31 +6,46 @@ def self.help
...
+  def initialize(cache_location, cleanup_outdated, cask_tokens="")
     @cache_location = Pathname(cache_location)
     @cleanup_outdated = cleanup_outdated
+    @cask_tokens = cask_tokens
   end
 
-  def cleanup!
-    remove_dead_symlinks
-    remove_all_cache_files
+  def cleanup!()
```
Comment (jawshooah) [inline lib/hbc/cli/cleanup.rb]: No need for empty parens here. The `!` communicates that this is a side-effecting method.

File: lib/hbc/cli/cleanup.rb [hunk 2]
```diff
@@ -6,31 +6,46 @@ def self.help
...
   end
 
   def cache_symlinks
     cache_location.children.select(&:symlink?)
   end
 
-  def dead_symlinks
-    cache_symlinks.reject(&:exist?)
+  def filter_out_deletable(to_filter)
+    if !(cask_tokens.empty?)
```
Comment (jawshooah) [inline lib/hbc/cli/cleanup.rb]: This method is unnecessarily verbose. You could accomplish the same thing with:

```rb
def filter_out_deletable(to_filter)
  return to_filter if cask_tokens.empty?
  to_filter.select do |item|
    cask_tokens.any? { |token| File.basename(item).include?(token) }
  end
end
```

File: lib/hbc/cli/cleanup.rb [hunk 3]
```diff
@@ -6,31 +6,46 @@ def self.help
     "cleans up cached downloads and tracker symlinks"
   end
 
-  def self.run(*_ignored)
-    default.cleanup!
+  def self.run(*args)
+    default(cask_tokens_from(args)).cleanup!()
```
Comment (jawshooah) [inline lib/hbc/cli/cleanup.rb]: Where is this `cask_tokens_from` method defined?

**Comment by jawshooah** [comment]
You should add some tests to `spec/cask/cli/cleanup_spec.rb`.

**Comment by clone1612** [comment]
I just can't seem to get the hang of working with GIT, sorry guys :/ Closing this pull request

**Comment by jawshooah** [comment]
@clone1612 would you like some help with the rebase? Otherwise I can take over and use your commits as a baseline if you don't mind.

**Comment by clone1612** [comment]
Feel free to take over, I don't mind at all

**Pull request closed without merging**